src/deadline/job_attachments/api/manifest.py

```diff
@@ -0,0 +1,347 @@
+        )
+
+    if not manifests or len(manifests) == 0:
+        logger.echo("No manifest generated")
+        return None
+
+    # This is a hard failure, we are snapshotting 1 directory.
+    assert len(manifests) == 1
+    output_manifest = manifests[0].asset_manifest
+    assert output_manifest
+
+    # If this is a diff manifest, load the supplied manifest file.
+    if diff:
+        # Parse local manifest
+        with open(diff) as source_diff:
+            source_manifest_str = source_diff.read()
+            source_manifest = decode_manifest(source_manifest_str)
+
+        # Get the differences
+        changed_paths: list[str] = []
+        differences: list[tuple[FileStatus, BaseManifestPath]] = compare_manifest(
+            source_manifest, output_manifest
+        )
+        for diff_item in differences:
+            if diff_item[0] == FileStatus.MODIFIED or diff_item[0] == FileStatus.NEW:
+                full_diff_path = f"{root}/{diff_item[1].path}"
+                changed_paths.append(full_diff_path)
+                logger.echo(f"Found difference at: {full_diff_path}, Status: {diff_item[0]}")
+
+        # Since the files are already hashed, we can easily re-use has_attachments to remake a diff manifest.
+        diff_group = asset_manager.prepare_paths_for_upload(
+            input_paths=changed_paths, output_paths=[root], referenced_paths=[]
+        )
+        _, diff_manifests = api.hash_attachments(
+            asset_manager=asset_manager,
+            asset_groups=diff_group.asset_groups,
+            total_input_files=diff_group.total_input_files,
+            total_input_bytes=diff_group.total_input_bytes,
+            print_function_callback=logger.echo,
+            hashing_progress_callback=hash_callback_manager.callback,
+        )
+        output_manifest = diff_manifests[0].asset_manifest
+
+    # Write created manifest into local file, at the specified location at destination
+    if output_manifest is not None:
+
+        timestamp = datetime.datetime.now().strftime("%Y-%m-%dT%H-%M-%S")
+        manifest_name = name if name else root.replace("/", "_")
+        manifest_name = manifest_name[1:] if manifest_name[0] == "_" else manifest_name
+        manifest_name = f"{manifest_name}-{timestamp}.manifest"
+
+        local_manifest_file = Path(destination, manifest_name)
+        local_manifest_file.parent.mkdir(parents=True, exist_ok=True)
+        with open(local_manifest_file, "w") as file:
+            file.write(output_manifest.encode())
+
+        # Output results.
+        logger.echo(f"Manifest Generated at {destination}{manifest_name}\n")
+        return ManifestSnapshot(manifest=f"{destination}{manifest_name}")
+    else:
+        # No manifest generated.
+        logger.echo("No manifest generated")
+        return None
+
+
+def _manifest_diff(
+    manifest: str,
+    root: str,
+    glob: str,
+    logger: ClickLogger = ClickLogger(False),
+    pretty_print: bool = False,
```

Comment:
Since we plan to expose these functions as APIs, shall we do more validation on the input?